### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by vatai**
Had a late night meeting yesterday and then made a mess with git... so everything took a bit longer than intended...

Is this approach, ok?

One big difference compared to vecto is that the `run_module()` is "silent" by default. If we need it not to be silent in the future, we can add an extra `verbose` argument to the function (with `True` default parameter to keep compatibility with earlier use of the function).

### CODE REVIEW

**Comment by vatai** [comment]
Ahhh... and I see that the CI is failing because it can't find TF. We should install it? Just something like `apt-get install tensorflow`? right?

**Comment by undertherain** [comment]
It's a bit tricky. Simple answer would have been add to requirements.txt and install everything with pip from .travis.yaml

the tricky part is that we don't want tf to be a strickt dependency as some users might want to benchmark only against, say, pytorch

so next step is to have requirements.txt and say test_requirements.txt

**Comment by undertherain** [comment]
another issue is that installing everything every time will make tests on travis run super long
I really liked the approach of gitlab to this: you define the base docker image in which the tests are running and only code is updated on commits. though you can rebuild the image as well automcatically when say dockerfiles are touched in the pipeline

maybe something like that is possible for travis - but not sure

**Comment by vatai** [comment]
I think travis has something like cache/cached.

The tests themselves are ok? Right? (On my local machine they are not failing of course)

**Comment by vatai** [comment]
Is [this](https://docs.travis-ci.com/user/caching/#pip-cache) what we are looking foor?

File: test/test_tensorflow.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+logging.basicConfig(level=logging.DEBUG)
+
+
+class TensorflowTests(unittest.TestCase):
+    # TODO(vatai): Run it quietly
+    def setUp(self):
+        self.name = "benchmarker"
+        self.imgnet_args = [
+            "--framework=tensorflow",
+            "--problem_size=32",
```
Comment (undertherain) [inline test/test_tensorflow.py:14]: I'd make it even smaller. say 2/4

File: requirements.txt [hunk 2]
```diff
@@ -3,4 +3,5 @@ py-cpuinfo
 pydataset
 system_query[cpu, hdd, ram, swap]
 python_papi
-# install system_query[all] for GPU systems
\ No newline at end of file
+tensorflow-gpu
```
Comment (undertherain) [inline requirements.txt:6]: I don't think there's GPU on travis, so can install tensorflow without gpu
also let's move frameworks to test_requirements

**Comment by vatai** [comment]
The tests themselves take quite long by themselves. Maybe making it smaller will speed things up, but loading TF takes forever!!! I'll try a few things.

**Comment by vatai** [comment]
@undertherain Isn't this in `.travis.yml` a bit redundant?
```yml
  - python -m unittest discover
  - python -m coverage run --branch --source . -m unittest discover --verbose
```
The second line also executes the tests, no?

**Comment by undertherain** [comment]
virtual node on travis is much weaker then probably even your laptop CPU  - so even 16 images will take longer, I'm mostly worried about some limit on cpu cycles they might have there

running two times - yeah, I think one would be enough
maybe there was a reason I did it like that - but don't remember :D maybe just by mistake, let's remove first line

**Comment by vatai** [comment]
現在 testing of the tf-resnet and tf-vgg, including installing (from cache) tf AND torch takes about 10 min. We'll see how many cases we can fit. I'll write the pytorch tests too hope for the best.

**Comment by vatai** [comment]
I've found a bug (I think). Please pay extra attention to commit 2d2d5c4.

**Pull request merged**